collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2830,8 +2836,18 @@ protected void onStop() {
         super.onStop();
     }
 
-    private void sendSavedBroadcast() {
-        sendBroadcast(new Intent("org.odk.collect.android.FormSaved"));
+    void enqueueFinalizedForm() {
+        Constraints constraints = new Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
+                .build();
+
+        OneTimeWorkRequest autoSendWork =
+                new OneTimeWorkRequest.Builder(AutoSendWorker.class)
+                        .addTag(AutoSendWorker.TAG)
+                        .setConstraints(constraints)
+                        .build();
+
+        WorkManager.getInstance().beginUniqueWork(AutoSendWorker.TAG, ExistingWorkPolicy.KEEP, autoSendWork).enqueue();
```

Comment:
@lognaturel Enforcement takes place right here. An unique job will be enqueue only if one doesn't exist. Thus the usage of of the `ExistingWorkPolicy` that "keeps" a job that's already enqueued.

Reply:
Perfect, thank you!